Formula/resin-cli.rb

```diff
@@ -0,0 +1,19 @@
+require "language/node"
+
+class ResinCli < Formula
+  desc "The official resin.io CLI tool"
+  homepage "https://docs.resin.io/reference/cli/"
+  url "https://registry.npmjs.org/resin-cli/-/resin-cli-7.10.6.tgz"
+  sha256 "011f79bfb1a75ad077b82afdc42695ce6061a522ccb1671a4e6d47e3e39f0889"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "Logging in to resin.io", pipe_output("#{bin}/resin login")
```

Comment:
I have swapped to shell_output, and tested for the exit code. This now specifies fake credentials to force the process not to open stdin, and it passes. The error message goes to stderr, which shell_output doesn't give me access to because it uses backticks, so I have opted not to test for it.

I would have to introduce one of the Ruby implementations of popen to get access to stderr.  I see quite a few formulas have done this already. It's probably worth opening another issue to add an assertion or modify shell_output and pipe_output to use popen to capture stderr and then cleanup all of the formulas that are doing their own thing to get at stderr.